### DISCUSSION TITLE
Vertically Split Main Editor Window for Game/Script Side-By-Side

### DISCUSSION THREAD

**Discussion by shadow64** [godotengine/godot-proposals#8347] [Editor]
I tried to find someone mentioning this, but I don't think it's come up. So far everyone seems to want the script window to be floating. I'd actually love to keep it in the center, but be able to split that space vertically and have it on the left or right:

![image](https://github.com/godotengine/godot-proposals/assets/4566839/fd4c9e80-98ae-4371-9d95-b805b7a0b195)

Maybe this is already covered by the people wanting to be able to dock every window, but it didn't seem like it applied to the main area. As someone who deals with sprite animation, it would be nice to see the sprites and the code at the same time without having to break it into another entire window.

This would actually be semi-possible right now if the floating version of the script window could add docked items to the left or right, but I'm sure some people would hate that. Personally, I just find I rarely need all of the horizontal space I have in the middle for code OR the game, so I'd much rather be able to look at both without needing a floating dock.

And while the floating dock is nice, it does come with that issue. To get similar functionality (the code against the game window) you'd need to move all other docked items to the left or right, which kind of drives me nuts because it makes sense to have these things centered. I like the general design of the editor, I just would like to split the middle area.

Edit: For those who noticed and were wondering: 
<IMG SRC="https://github.com/godotengine/godot-proposals/assets/4566839/eb7fe650-ed1c-4aec-8b24-72bb97ccdbf1" width="200" />

(I did TRY to remove all mention of it! Haha.)

**Comment by azur-wolve** [comment]
this could be useful for ultra-widescreeens, but its a bit niche

**Reply by shadow64** [reply]
Hmm. Maybe it is. What were you thinking made it so?

Since the editor UI is already very modular, this is kind of the only way it isn't. Having the code and the game window forcibly share the same space is (I think) the reason we got the pop-out option - I just don't like that solution because it's putting the code further away from the center, where my eyes already are - honestly I'd rather be able to pop out the game/scene window than the scripting area.

I'm on a 1440p monitor myself, but you're right: I think ultra-widescreen users would use it.

**Reply by aXu-AP** [reply]
I don't have an ultra-wide, but my screen is 4k and I could use this with ease. Script editor doesn't need much space horizontally (if you keep your lines under 100 characters long).

**Reply by coffandro** [reply]
Personally It'd find it useful even though I just have a laptop screen, amazing for refactoring old code and such. you can have the old code or documentation on the left and write on the right.

**Reply by themipper** [reply]
I would love to have the ability to split and freely move the different windows, tabs and docks within the editor.

I have an ultrawide and the screen space is just not used properly for some scenarios. And as mentioned you don't need an ultra-wide for this to be true.

Like comparing two script files, comparing two scenes or having the scene on one side and the script on the other.

You can achieve something similar by making the script dock float and dock that window side by side to the editor window. You can also move the right or left hand side docks to the other side. So that only one side of the window is cluttered by these docks (I love having Scene and Inspector docks side by side - less mouse movement needed). Then the end-result looks similar to what you want to have.

![2025-06-23 14_34_38-Script Editor - Godot Engine](https://github.com/user-attachments/assets/9b8d866e-6154-42b6-95a1-f27ed3751e69)

**Reply by mistaherd** [reply]
this woukld be so handy for different scenes

**Comment by AndreaMonzini** [comment]
Hello,
actually i was searching a solution for this feature.

I use a single 32" 4k Monitor and i think it would be very useful in my workflow to work on script the editor and on the 2D/3D scene side by side.

I know that i can make the script editor floating but i am not able to use the drag and drop functionality from the "Scene" and "filesystem".
Also i would use the Script editor on the left beside the node "Scene" tab,  and with the 2D/3D Scene on the right.
With the floating script window option this configuration is not possible.

So i think it could be useful not only for widescreen monitor but also for who prefer to work with a single and high resolution monitor.

Maybe it could be considered in the future and probably the possibility to configure every window like in Blender would be optimal to maximize the personalization of the spaces.

I tried a quick mockup but of course the layout space should be more optimized ( like resize vertically the 3D scene ) for this feature:


<img width="3832" height="2022" alt="script_side_by_side" src="https://github.com/user-attachments/assets/02929640-9d33-4420-9ae6-fc37d3f1dfcc" />

**Reply by SourCandyLime** [reply]
This! This is exactly what I need! The way I am making my game, it gets wonked up by the main renderer on purpose, so if I could have my editor scene open and the game scene open in split tandum that would be GLORIOUS!
